Created the end game portal

brandon722 opened the pull request:
## Description

This pull request adds a portal visual effect to a cube in the game. The purpose of the portal is the end game mechanic. Once all minigames have been completed, the GameObject will be set to active, allowing the player to walk through the portal and win the game. 

## Type of Change

- [ ] Bug Fix
- [x] New Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] Documentation
- [ ] Other (please specify):

## Related Issue

Link to the issue this pull request addresses (if applicable):
- N/A

## Changes Made
- Created a new Visual Effect Graph for the portal effect.
- Attached the portal visual effect to a cube GameObject.
- Attached the scripts for managing the portal effect and its interactions.
    - Attached the portal GameObject to the MinigameManager.
- Updated the Map Core prefab to include the new portal effect.
- Added the EndCredits scene to the build profiles.

## Checklist

- [x] Code follows the project's coding style.
- [x] If asset(s) added, they have been made into a prefab
- [ ] Tests have been added/updated.
- [x] Documentation has been updated (if applicable).
- [x] No unused code or assets remain.
- [x] Unity project builds without errors.

## Screenshots (if applicable)

https://github.com/user-attachments/assets/0599ee66-ef7a-48c9-a625-e5176a7eef67

## Additional Notes

I tried my best to orient the portal inside of our portal generator. Sometimes, the portal effects bleed out of the portal, but for now, I think this looks pretty good. If editing the portal, do not increase the particle size past 0.1, as I have noticed it will create massive lag for the player.

brandon722 committed:
Created portal visual effects and attached it to a cube.

brandon722 committed:
Changed portal color to blue and better matched the particle sizes to the portal generator.

brandon722 committed:
Added EndCredits scene to the build profiles and attached the portal collision script to the portal.

brandon722 committed:
Updated the Map Core prefab.

uid106 reviewed (approved):
## Pull Request Review
- **Functionality**:
  - Added graphical effect for portal 
  - Attached portal to gameobject
  - Attached script for opening portal when all tasks are completed
  - Updated map core

- **Unused Code**: 
  - None

- **Code Style**: 
   - Properly following projects code style 

- **Script Structure**: 
   -  The structure of scripts is appropriate

- **Documentation**: 
   - Follows project documentation style

- **Build Integrity**: 
  - The project builds and runs as intended

---

### Summary
Tested game completion and portal will spawn in when all tasks are completed. Portal functioned in bringing up the end credit screen. The particle effects look great. PR is ready to merge.

EricPettie reviewed (approved):
## Pull Request Review
- **Functionality**:
  - Added end game portal
  - Updated map
  
- **Unused Code**: 
  - None

- **Code Style**: 
   - Aligns with the project

- **Script Structure**: 
   -  Everything is organized

- **Documentation**: 
   - The comments are clear and describe the code well

- **Build Integrity**: 
  - The game successfully built and ran

---

### Summary
Portal successfully spawns in when tasks are completed. The portal also looks fantastic. Looks good to merge.

uid106 committed:
Updated completion items & map

Added map into scene and can now turn on with M key. Added green particle effects on generator upon completion. Added green lights around portal upon completion.
